reporter.go

```diff
@@ -0,0 +1,128 @@
+package check
+
+import (
+	"fmt"
+	"io"
+	"sync"
+)
+
+type reporter interface {
```

Comment:
This is a huge interface, does all this functionality need to be exposed? If it is used in more than one place then maybe we could split it in several interfaces. If all the methods are used in only one place then perhaps that logic should be moved into this type